### ISSUE TITLE
NVDA+Up fails to read or spell many controls and lines in Google Chrome

### ISSUE THREAD

**Issue by amirsol81** [nvaccess/nvda#11285]
Insert+Up (NVDA+Up) is used in NVDA to report the current line under the application cursor. Pressing it twice will spell the current line. Pressing three times will spell the line using character descriptions. However, pressing it on many controls in Google Chrome 127 forces NVDA to remain silent instead of reporting the current line or spelling the current line. This happens in various places like the menu (ALT+F), the list of bookmarks (ALT+F+B), etc.
Note: the issue was originally reported with Chrome 83, but it's duplicable with Chrome 127.
### Steps to reproduce:
1. Open Google Chrome and press Control+D to create a bookmark.
2. Press Tab or SHIFT+Tab to move around various controls in the Bookmark creation dialog and press Insert+Up (NVDA+Up) on each control, once or twice.
3. Alternatively, you can simply press ALT+F in Google Chrome and try pressing Insert+Up on items such as History submenu, Bookmarks submenu, Make text smaller, Make text larger, Full screen, Find, More tools, etc.
4. You may also try pressing Insert+Up on your Bookmarks in the Bookmarks submenu.
### Actual behavior:
For most controls and items above, using Insert+Up forces NVDA to remain silent.
### Expected behavior:
For these controls and items, NVDA should report the current line under the application cursor or spell it.
#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
Alpha 20381, but also duplicable with alphas going back to many months. Duplicable with Alpha 33355. Also duplicable with 2020.1 to 2024.3 RC1.
#### Windows version:
Windows 10 Pro 64-Bit, version 2004, and Windows 11 24H2 (26100.1457)
#### Name and version of other software in use when reproducing the issue:
Google Chrome 83 Official to Chrome 127.
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Various releases over the past 3 years.
#### If addons are disabled, is your problem still occuring?
Yes
#### Did you try to run the COM registry fixing tool in NVDA menu / tools?
Yes

**Comment by amirsol81** [comment]
It is also worth mentioning that one can use Insert+NumPad5 (NVDA+NumPad5) as a work-around to read or spell controls or items on which Insert+Up only says Object Replacement Characte
r.

**Comment by DrSooom** [comment]
Sounds similar to issue #10634 and it's indirectly related to issue #11177.

**Comment by Adriani90** [comment]
This is a duplicate of #11177. Even if that issue title points to Firefox, the discussion goes further and touches also other places where object replacement characters are announced (i.e. Thunderbird). I am closing this one as duplicate.

**Comment by Adriani90** [comment]
Ok it seems I was wrong here, this issue persists for a longer time. I thought it was related to the object replacement character. In fact, I tested with NVDA 2019.3.1 and the current line is not reported at all on these objects.

**Comment by Adriani90** [comment]
This issue even occurs with NVDA 2018.2.1. @amirsol81 could you please test with an older version of Chrome?
It works just well in Microsoft Edge Chromium or Firefox. So this seem really Chrome specific.
cc: @ObjectInSpace

**Comment by Adriani90** [comment]
Ah and if you don't mind, could you please change the title and the description saying that NVDA does not report the current line? It can be confusing for testers or developers as it is now, since the behavior changed in last alpha. Thanks.

**Comment by ObjectInSpace** [comment]
I can confirm that the current line is not read in Google Chrome for NVDA 2020.1 in Chrome 85, but it reads correctly for Chrome 83. So I think this issue you are now reporting is unrelated to the original one. In any case, can you confirm that this does not occur in Chrome 83.0.4103.106? I'll file this Chrome-side. Thanks much!

**Comment by amirsol81** [comment]
@ObjectInSpace You are right. However, since I heard "Object Replacement Character" instead of the text of the actual control, I thought it's related to the one frequently discussed here.
This also happens with NVDA Official release and Chrome 85.

**Comment by amirsol81** [comment]
@Adriani90 I updated the issue title and description. It's worth mentioning that I'm running Google Chrome version 83.0.4103.116 and don't have access to older releases/Chrome betas for testing purposes.

**Comment by ObjectInSpace** [comment]
It doesn't occur in M84 either so this is almost certainly a Chrome regression. Filed https://crbug.com/1098528 to make sure it gets taken care of.

**Comment by ObjectInSpace** [comment]
To be clear, the bug I filed is about NVDA+up being silent. I haven't looked into the object-replacement-character bug this was duped against yet as it seems to be an issue being worked on in NVDA alpha. LMK if I got that wrong.

**Comment by amirsol81** [comment]
> To be clear, the bug I filed is about NVDA+up being silent. I haven't looked into the object-replacement-character bug this was duped against yet as it seems to be an issue being worked on in NVDA alpha. LMK if I got that wrong.

@ObjectInSpace You are right. That's why I updated the bug title and description. I also added Chrome 83 to the title as I don't have other Chrome releases, old or beta.

**Comment by Adriani90** [comment]
@ObjectInSpace it seems the Chrome bug is fixed. In which Chrome version will it be visible? I am on Canari 86 ut the issue is still reproducible there.

**Comment by amirsol81** [comment]
It's still fully reproduceable with Google Chrome 84, the official release.

**Comment by ObjectInSpace** [comment]
Thanks for the pointer! Looks like this was only partially fixed. In M84 and above, you should be able to read the currently focused menu item or submenu item by line. What is not currently working are items in history or bookmarks or the tab switcher. It's not clear to me if this ever worked, however. I tested versions of Chrome that were significantly earlier than M83 and some of them read nothing at all. Updated the crbug to reflect the current status.

**Comment by amirsol81** [comment]
This is still duplicable with Google Chrome version 85.0.4183.83 and latest NVDA alpha in the History/Bookmarks/Tab switcher areas.

**Comment by amirsol81** [comment]
With Chrome 89 and the latest NVDA Alpha, this just affects items in the History list. But items in the menu and Bookmarks can once again be spelled.

**Comment by Adriani90** [comment]
This issue is not reproducible anymore with NVDA alpha-23579,07f7bf1c and Chrome 92 under windows 
10 pro 21 H1
Closing as works for me.

**Comment by amirsol81** [comment]
@Adriani90 @Qchristensen @seanbudd Please reopen this as the issue is perfectly duplicable with recent releases of Chrome and all recent releases of NVDA. In fact, it hasn't been really fixed according to my experience, so I'm wondering why this has been closed. NVDA still fails to report most Chrome controls with NVDA+Up, and the same is true about spelling them with NVDA+Up pressed twice.

**Comment by amirsol81** [comment]
@seanbudd @Adriani90 @Qchristensen I'm tagging you as this issue hasn't been triaged yet. It affects Chrome V127 and the latest NVDA alphas/beta/official release.

**Comment by DrSooom** [comment]
The latest stable version of NVDA currently is [2024.2](https://www.nvaccess.org/files/nvda/releases/2024.2/nvda_2024.2.exe) and the latest Beta version [2024.3 Beta 6](https://www.nvaccess.org/files/nvda/releases/2024.3beta6/nvda_2024.3beta6.exe).

**Comment by seanbudd** [comment]
@amirsol81 - can you please update the issue to reflect the current state of the issue

**Comment by Qchristensen** [comment]
I can replicate @amirsol81's original scenario with the bookmark popup dialog.  Note that this is a browser dialog and nothing on a web page itself.

NVDA 2024.3 RC1
Windows 11 (64-bit) Version: 23H2, Build: 22631.3958 
Chrome Version: 127.0.6533.100 

Note that a workaround is to press NVDA+tab which does correctly read the control with focus.

**Comment by amirsol81** [comment]
@seanbudd Additionally, NVDA+Up fails to read, or spell, menu items as 
we press ALT to open Chrome's menus. For instance, the list of bookmarks 
we've already generated can't be read or spelled via NVDA+Up though 
NVDA+TAB reads them. This is with Chrome 127.

**Comment by seanbudd** [comment]
@amirsol81 - thank you for the information as a comment, but can you please update the original issue description with the latest information.

**Comment by amirsol81** [comment]
@seanbudd Sure, I just updated the issue description.

**Comment by jcsteh** [comment]
@amirsol81, can you still reproduce this? I can't reproduce it in Chrome Canary: Version 139.0.7256.0 (Official Build) canary (64-bit).

**Comment by amirsol81** [comment]
> [@amirsol81](https://github.com/amirsol81), can you still reproduce this? I can't reproduce it in Chrome Canary: Version 139.0.7256.0 (Official Build) canary (64-bit).

@jcsteh It's still duplicable with Google Chrome version 137.0.7151.120. I'm not testing the 138 or 139 builds, though.

**Comment by jcsteh** [comment]
Hmm. I wonder whether native UIA isn't enabled in stable yet, which might impact this.

@amirsol81, if you have the time and inclination, would you mind testing [this build](https://ci.appveyor.com/api/buildjobs/04ttu83k163ccvqn/artifacts/output%2Fnvda_snapshot_pr18271-36902%2C211960c7.exe) from PR #18271 to see if it fixes the problem? A temporary or portable copy will be fine. The fact that it's fixed in 139 suggests it's going to be fixed soon enough in Chrome itself anyway, so it doesn't really matter. That said, if it's fixed by that PR, I may as well reference it as fixed there.

**Comment by amirsol81** [comment]
> [@amirsol81](https://github.com/amirsol81), if you have the time and inclination, would you mind testing [this build](https://ci.appveyor.com/api/buildjobs/04ttu83k163ccvqn/artifacts/output%2Fnvda_snapshot_pr18271-36902%2C211960c7.exe) from PR [#18271](https://github.com/nvaccess/nvda/pull/18271) to see if it fixes the problem? A temporary or portable copy will be fine. The fact that it's fixed in 139 suggests it's going to be fixed soon enough in Chrome itself anyway, so it doesn't really matter. That said, if it's fixed by that PR, I may as well reference it as fixed there.

@jcsteh I installed a portable copy of the try build, and it fixes the issue with Chrome 137. Thanks for taking a stab at it!

**Comment by jcsteh** [comment]
It's actually a happy accident. I didn't intend to fix this particular bug at all. :)

**Comment by jcsteh** [comment]
@seanbudd Oops, sorry, I think my comment in #20096 unintentionally triggered an auto close here. Emphasis mine:

> The narrow scoping here means that this probably doesn't *fix #11285* in the Chrome UI. I was never able to reproduce it, so I can't verify that. Regardless, I think that should be addressed separately, though the new property should make that much simpler.

It's possible this *is* fixed, but I doubt it.

**Comment by seanbudd** [comment]
@amirsol81 - can you test and see if this is still an issue

**Comment by amirsol81** [comment]
@seanbudd I'm running NVDA alpha-55911 and Google Chrome V148, and this issue can no longer be duplicated at all. Everything can be read or spelled via NVDA+Up pressed once or twice. So the issue can be safely closed.

**Comment by amirsol81** [comment]
@seanbudd @jcsteh Unfortunately this issue hasn't been fully resolved. I just noticed that NVDA, both latest alpha and latest official release, can't read or spell the items in the Chrome history. Simply press Control+H, press Tab a couple of times to reach the list of items, and try pressing NVDA+Up once, and twice, on them. NVDA reads nothing,

**Comment by amirsol81** [comment]
@seanbudd @jcsteh So I'm re-opening this issue. It can be duplicated with Chrome 146 to 149. For refrence, JAWS can read the History items with Insert+Up. The History items are apparently inside a table.

**Comment by Qchristensen** [comment]
I haven't looked into the reasons why (whether it is poor coding of the Chrome History page for instance), and I can confirm the same behaviour, in Chrom as well as in Edge.

BUT, I did find two workarounds for anyone impacted by this:

1) Press NVDA+tab which essentially reads the current line on Google's history page.  Press twice to spell.
2) In NVDA 2026.2 (betas currently) you can press NVDA+x to repeat the last utterance.  Press twice to open in a browsable dialog you can move through by letter / copy from etc).

**Comment by hwf1324** [comment]
I think this has something to do with `IA2TextTextInfo`.

**Comment by jcsteh** [comment]
The issue on the Chrome History page is the same as #15159. I've made multiple attempts to fix this, but it keeps having to be reverted for various reasons. But anyway, fixing that will also fix the History page issue or any other case where browse mode can be engaged. It won't fix any case where browse mode cannot be engaged.
